aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecorator.kt

```diff
@@ -136,89 +156,99 @@ class SmokeTestsBuilderKindBehavior(val codegenContext: CodegenContext) : Instan
+
+    fun render(
         writer: RustWriter,
         testCase: SmokeTestCase,
-    ) {
-        writer.rust(
-            "let config = #{T}::load_defaults(config::BehaviorVersion::latest()).await;",
-            AwsCargoDependency.awsConfig(codegenContext.runtimeConfig).toType(),
-        )
-        writer.rust("let conf = config::Config::from(&config).to_builder()")
-        writer.indent()
+    ) = writer.apply {
+        renderConf(this, testCase)
+        rust("let client = Client::from_conf(conf);")
+        renderInput(this, testCase.params)
+        renderExpectation(this, testCase.expectation)
+    }
+
+    private fun renderConf(
+        writer: RustWriter,
+        testCase: SmokeTestCase,
+    ) = writer.apply {
+        rustTemplate("#{config_builder_initializer}", "config_builder_initializer" to configBuilderInitializer())
+        indent()
 
         // TODO(https://github.com/smithy-lang/smithy-rs/issues/3776) Once Account ID routing is supported,
         //  reflect the config setting here, especially to disable it if needed, as it is enabled by default in
         //  `AwsVendorParams`.
 
         val vendorParams = AwsSmokeTestModel.getAwsVendorParams(testCase)
         vendorParams.orNull()?.let { params ->
-            writer.rust(".region(config::Region::new(${params.region.dq()}))")
-            writer.rust(".use_dual_stack(${params.useDualstack()})")
-            writer.rust(".use_fips(${params.useFips()})")
-            params.uri.orNull()?.let { writer.rust(".endpoint_url($it)") }
+            rust(".region(config::Region::new(${params.region.dq()}))")
+            rust(".use_dual_stack(${params.useDualstack()})")
+            rust(".use_fips(${params.useFips()})")
+            params.uri.orNull()?.let { rust(".endpoint_url($it)") }
         }
 
         val s3VendorParams = AwsSmokeTestModel.getS3VendorParams(testCase)
         s3VendorParams.orNull()?.let { params ->
-            writer.rust(".accelerate_(${params.useAccelerate()})")
-            writer.rust(".force_path_style_(${params.forcePathStyle()})")
-            writer.rust(".use_arn_region(${params.useArnRegion()})")
-            writer.rust(".disable_multi_region_access_points(${params.useMultiRegionAccessPoints().not()})")
+            rust(".accelerate_(${params.useAccelerate()})")
+            rust(".force_path_style_(${params.forcePathStyle()})")
+            rust(".use_arn_region(${params.useArnRegion()})")
+            rust(".disable_multi_region_access_points(${params.useMultiRegionAccessPoints().not()})")
         }
 
-        writer.rust(".build();")
-        writer.dedent()
+        rust(".build();")
+        dedent()
     }
 
-    fun renderInput(
+    private fun renderInput(
         writer: RustWriter,
-        operationShape: OperationShape,
-        inputShape: StructureShape,
         data: Optional<ObjectNode>,
         headers: Map<String, String> = mapOf(),
         ctx: Ctx = Ctx(),
-    ) {
+    ) = writer.apply {
```

Comment:
TIL about `.apply`. I love this.

Reply:
[Learned from](https://github.com/smithy-lang/smithy-rs/pull/3662#discussion_r1613795127) @drganjoo 😁